test/integration/clustercidr/ipam_test.go

```diff
@@ -138,6 +139,456 @@ func TestIPAMMultiCIDRRangeAllocatorType(t *testing.T) {
+
+	clientSet := clientset.NewForConfigOrDie(kubeConfig)
+	sharedInformer := informers.NewSharedInformerFactory(clientSet, 1*time.Hour)
+
+	ipamController := booststrapMultiCIDRRangeAllocator(t, clientSet, sharedInformer)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go ipamController.Run(ctx.Done())
+	sharedInformer.Start(ctx.Done())
+
+	t.Run("delete cc with node associated", func(t *testing.T) {
+
+		// Create a ClusterCIDR.
+		clusterCIDR := makeClusterCIDR("dualstack-cc-del", "192.168.0.0/23", "fd00:30:100::/119", 8, nodeSelector(map[string][]string{"ipv4": {"true"}, "ipv6": {"true"}}))
+		if _, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Create(context.TODO(), clusterCIDR, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		// Wait for the ClusterCIDR to be created.
+		if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
+			cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), clusterCIDR.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			return cc != nil, nil
+		}); err != nil {
+			t.Fatalf("failed while waiting for ClusterCIDR %q to be created: %v", clusterCIDR.Name, err)
+		}
+
+		// Sleep for one second to make sure the controller processes the newly created ClusterCIDR.
+		time.Sleep(1 * time.Second)
+
+		// Create a node, which gets pod CIDR from the clusterCIDR created above.
+		node := makeNode("dualstack-node", map[string]string{"ipv4": "true", "ipv6": "true"})
+		expectedPodCIDRs := []string{"192.168.0.0/24", "fd00:30:100::/120"}
+		if _, err := clientSet.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+		if gotPodCIDRs, err := nodePodCIDRs(clientSet, node.Name); err != nil {
+			t.Fatal(err)
+		} else if !reflect.DeepEqual(expectedPodCIDRs, gotPodCIDRs) {
+			t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", expectedPodCIDRs, gotPodCIDRs)
+		}
+
+		// Delete the ClusterCIDR
+		if err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Delete(context.TODO(), clusterCIDR.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		//Wait and make sure that the ClusterCIDR is not deleted, as there is a node associated with it.
+		if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
+			cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), clusterCIDR.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			return cc != nil, nil
+		}); err != nil {
+			t.Fatalf("failed while waiting for ClusterCIDR %q to be deleted: %v", clusterCIDR.Name, err)
+		}
+
+		//Delete the node.
+		if err := clientSet.CoreV1().Nodes().Delete(context.TODO(), node.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		// Poll to make sure that the Node is deleted.
+		if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
```

Comment:
https://github.com/kubernetes/kubernetes/pull/115291/commits/1e682f1fdab179171b08535eab8f778380596753#r1089824131

Reply:
Didn't get this? Is it for the timeout? If so I have updated this to `wait.ForeverTestTimeout` as well